test/parallel/test-https-agent-create-connection.js

```diff
@@ -132,3 +132,26 @@ function createServer() {
     }));
   }));
 }
+
+// `options` should not be modified
+{
+  const server = createServer();
+  server.listen(0, common.mustCall(() => {
+    const port = server.address().port;
+    const host = 'localhost';
+    const options = {
+      port: 3000,
+      rejectUnauthorized: false
+    };
+
+    const socket = agent.createConnection(port, host, options);
+    socket.on('connect', common.mustCall((data) => {
+      socket.end();
+    }));
+    socket.on('end', common.mustCall(() => {
+      assert.strictEqual(options.port, 3000);
+      assert.strictEqual(options.host, undefined);
```

Comment:
Nit: these two checks could be replaced with a deep equal check as in:

```js
      assert.deepStrictEqual(
        options,
        { port: 3000, rejectUnauthorized: false }
      );
```

Reply:
The `deepStrictEqual` is actually correct because my approach only checks for `port` and `host` keys. But in future, there may be a possibility that other props get added. So, the current test would fail for it but using `deepStrictEqual` would not fail the test. I believe it is not a Nit but important.